pkg/deploy/elbv2/target_group_synthesizer_test.go

```diff
@@ -0,0 +1,625 @@
+							TargetGroupArn: awssdk.String("arn-1"),
+						},
+						Tags: map[string]string{
+							"ingress.k8s.aws/resource": "id-1",
+						},
+					},
+					{
+						TargetGroup: &elbv2sdk.TargetGroup{
+							TargetGroupArn: awssdk.String("arn-2A"),
+						},
+						Tags: map[string]string{
+							"ingress.k8s.aws/resource": "id-2",
+						},
+					},
+					{
+						TargetGroup: &elbv2sdk.TargetGroup{
+							TargetGroupArn: awssdk.String("arn-2B"),
+						},
+						Tags: map[string]string{
+							"ingress.k8s.aws/resource": "id-2",
+						},
+					},
+				},
+				resourceIDTagKey: "ingress.k8s.aws/resource",
+			},
+			want: map[string][]TargetGroupWithTags{
+				"id-1": {
+					{
+						TargetGroup: &elbv2sdk.TargetGroup{
+							TargetGroupArn: awssdk.String("arn-1"),
+						},
+						Tags: map[string]string{
+							"ingress.k8s.aws/resource": "id-1",
+						},
+					},
+				},
+				"id-2": {
+					{
+						TargetGroup: &elbv2sdk.TargetGroup{
+							TargetGroupArn: awssdk.String("arn-2A"),
+						},
+						Tags: map[string]string{
+							"ingress.k8s.aws/resource": "id-2",
+						},
+					},
+					{
+						TargetGroup: &elbv2sdk.TargetGroup{
+							TargetGroupArn: awssdk.String("arn-2B"),
+						},
+						Tags: map[string]string{
+							"ingress.k8s.aws/resource": "id-2",
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "targetGroups without resourceID tag",
+			args: args{
+				sdkTGs: []TargetGroupWithTags{
+					{
+						TargetGroup: &elbv2sdk.TargetGroup{
+							TargetGroupArn: awssdk.String("arn-1"),
+						},
+						Tags: map[string]string{},
+					},
+				},
+				resourceIDTagKey: "ingress.k8s.aws/resource",
+			},
+			wantErr: errors.New("unexpected targetGroup with no resourceID: arn-1"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := mapSDKTargetGroupByResourceID(tt.args.sdkTGs, tt.args.resourceIDTagKey)
+			if tt.wantErr != nil {
+				assert.EqualError(t, err, tt.wantErr.Error())
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, tt.want, got)
+			}
+		})
+	}
+}
+
+func Test_isSDKTargetGroupRequiresReplacement(t *testing.T) {
+	type args struct {
+		sdkTG TargetGroupWithTags
+		resTG *elbv2model.TargetGroup
+	}
+	tests := []struct {
+		name string
+		args args
+		want bool
+	}{
+		{
+			name: "targetGroup don't need replacement",
+			args: args{
+				sdkTG: TargetGroupWithTags{
+					TargetGroup: &elbv2sdk.TargetGroup{
+						TargetType:      awssdk.String("ip"),
+						Port:            awssdk.Int64(8080),
+						Protocol:        awssdk.String("HTTP"),
+						TargetGroupName: awssdk.String("my-tg"),
+					},
+				},
+				resTG: &elbv2model.TargetGroup{
+					Spec: elbv2model.TargetGroupSpec{
+						TargetType: elbv2model.TargetTypeIP,
+						Port:       8080,
+						Protocol:   elbv2model.ProtocolHTTP,
+						Name:       "my-tg",
+					},
+				},
+			},
+			want: false,
+		},
+		{
+			name: "name-only change shouldn't need replacement",
```

Comment:
Does this apply for all target group protocols? Or it is specific to ALB?

Reply:
it applies to all protocols.